load.php

```diff
@@ -108,7 +108,23 @@ function perflab_get_module_settings() {
 	// Even though a default value is registered for this setting, the default must be explicitly
 	// passed here, to support scenarios where this function is called before the 'init' action,
 	// for example when loading the active modules.
-	return (array) get_option( PERFLAB_MODULES_SETTING, perflab_get_modules_setting_default() );
+	$perflab_get_module_settings = (array) get_option( PERFLAB_MODULES_SETTING, perflab_get_modules_setting_default() );
+
+	$legacy_module_slug = array(
+		'site-health/audit-autoloaded-options' => 'database/audit-autoloaded-options',
+		'site-health/audit-enqueued-assets'    => 'css-and-js/audit-enqueued-assets',
+		'site-health/audit-full-page-cache'    => 'object-cache/audit-full-page-cache',
+		'site-health/webp-support'             => 'images/webp-support',
+	);
+
+	foreach ( $perflab_get_module_settings as $module_slug => $module_data ) {
+		if ( isset( $legacy_module_slug[ $module_slug ] ) ) {
+			$perflab_get_module_settings[ $legacy_module_slug[ $module_slug ] ] = $module_data;
+			unset( $perflab_get_module_settings[ $module_slug ] );
+		}
+	}
+
+	return $perflab_get_module_settings;
 }
```

Comment:
Can you please add test coverage for this change? We should add at least one test that sets the legacy module slug in the DB, and then verify that this function still returns the same data for the new module slug.

We could use a `dataProvider` for that test in order to cover all 4 migrations.

Reply:
Thank you for the feedback Felix. Unit test added https://github.com/WordPress/performance/pull/566/commits/ce593f4c9a629cd59cf4ad5d053d819569f44b5e